api/client/copy.go

```diff
@@ -0,0 +1,106 @@
+package client
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/docker/docker/pkg/archive"
+)
+
+// CmdCp handles copying files or directories to/from/accross containers.
+func (cli *DockerCli) CmdCp(args ...string) error {
+	cmd := cli.Subcmd("cp", "[SRCCONTAINER:]SRCPATH [DSTCONTAINER:]DSTPATH", "Copy files/folders from SRCPATH to the DSTPATH")
+	if err := cmd.Parse(args); err != nil {
+		return nil
+	}
+
+	if cmd.NArg() != 2 {
+		cmd.Usage()
+		return nil
+	}
+
+	var (
+		srcPath      = cmd.Arg(0)
+		dstPath      = cmd.Arg(1)
+		infoSrc      = strings.SplitN(srcPath, ":", 2)
+		infoDst      = strings.SplitN(dstPath, ":", 2)
+		srcContainer string
+		dstContainer string
+	)
+
+	if len(infoSrc) != 2 && len(infoDst) != 2 {
+		// User didn't specify any container.
+		return fmt.Errorf("error: Wrong path format. Must specify a source and/or destination container")
+	}
+
+	if len(infoSrc) == 2 {
+		// Copy from a container.
+		srcContainer, srcPath = infoSrc[0], infoSrc[1]
+	}
+	if len(infoDst) == 2 {
+		// Copy to a containers.
+		dstContainer, dstPath = infoDst[0], infoDst[1]
```

Comment:
with all the code above this, consider a `direction` flag and `src` and `dst` variables instead of what you are doing now. I think the code will be cleaner and more obvious, and simpler.